modules/rs_grading.py

```diff
@@ -441,21 +441,29 @@ def _compute_assignment_total(student, assignment, course_name, db=None):
                          .format(student.id, assignment.id, score))
         return score, None
 
-def _get_students(course_id, sid = None, db=None):
-    if sid:
-        # sid which is passed in is a row id, not a username; get the username
-        student_rows = db((db.user_courses.course_id == course_id) &
-                          (db.user_courses.user_id == db.auth_user.id) &
-                          (db.auth_user.username == sid)
+def _get_students(course_id=None, sid = None, student_rownum=None, db=None):
+    print("_get_students", course_id, sid, student_rownum)
+    if student_rownum:
+        # get the student id as well as username
+        student_rows = db((db.auth_user.id == student_rownum)
                           ).select(db.auth_user.username, db.auth_user.id)
-    else:
+    elif sid:
+        # sid is a username; get the matching rownum as well
+        # THIS IS DANGEROUS, if username is duplicated across courses. OK if username is unique server-wide
```

Comment:
Usernames are unique server wide, enforced by the database

Reply:
OK. I will remove the comment about it being dangerous.